src/main/java/no/unit/nva/cristin/institutions/Nester.java

```diff
@@ -0,0 +1,84 @@
+package no.unit.nva.cristin.institutions;
+
+import no.unit.nva.cristin.institutions.model.Identifier;
+import no.unit.nva.cristin.institutions.model.UnitObject;
+import no.unit.nva.cristin.institutions.model.UnitObjectBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+
+public class Nester {
+
+    public static final String MATCH_DOT = "\\.";
+    public static final int UNSPECIFIED_LEVEL = 0;
+    public static final int SECTION_LEVEL = 3;
+    public static final int DEPARTMENT_LEVEL = 2;
+    public static final int FACULTY_LEVEL = 1;
+    private transient UnitObject unitObject;
+
+    /**
+     * A method to take a flat array of Units and nest them according to their identifiers,
+     * so that 123.1.1.1 is nested under 123.1.1.0, which is nested under 123.1.0.0 and so on.
+     *
+     * @param unitObjects       An array of UnitObject objects
+     * @param subunitIdentifier In cases where only parent unitObjects are to be returned, pass the subunit identifier
+     */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public Nester(UnitObject[] unitObjects, Identifier subunitIdentifier) {
```

Comment:
The constructor needs to be tested. Indicatively, we need to test for when we have only first level children, only second level children and third level children. We need to test with one and more than one children